### ISSUE TITLE
Terraform providers for components issue

### ISSUE THREAD

**Issue by yevhenii0bozhenko** [labd/mach-composer#232]
Hi!
We where using mach-composer docker image in GitHub workflow with version 1.1
But the recent updates to our components require the terraform version >0.14.8:
```
Error: Incompatible terraform version

  on .terraform/modules/***/terraform/kuber-resources.tf line 201, in resource "kubernetes_manifest" "secretproviderclass":
 201: resource "kubernetes_manifest" "secretproviderclass" {

The `kubernetes_manifest` resource requires Terraform v0.14.8 or above
```
So i have decided to use [this](https://github.com/labd/mach-composer/blob/main/docs/src/howto/ci/github.md) approach instead, installing a particular version of terraform and mach-composer.
Unfortunately i cant install mach-composer version 1.1 using github action "mach-composer/setup-mach-composer@main":
```
Run mach-composer/setup-mach-composer@main
  with:
    version: 1.1
  env:
    MACH_COMPOSER_VERSION: 1.1
    CONFIG_FILE: main.yml
    TERRAFORM_CLI_PATH: /home/runner/work/_temp/***
  
Error: Unexpected HTTP response: 404
```
So i'm trying with the new version: 2.5.4
But it gives me another error:
```
╷
│ Error: Module module.algolia contains provider configuration
│ 
│ Providers cannot be configured within modules using count, for_each or
│ depends_on.
╵
```
Algolia component has indeed provider specified in .tf file like this:
```
terraform {
  required_providers {
    algolia = {
      source  = "k-yomo/algolia"
      version = "0.5.6"
    }
  }
}

provider "algolia" {
  app_id  = var.secrets.blabla
  api_key = var.secrets.bla
}
```

The question is:
How to specify provider information the way mach-composer with version 2.5.4 or above could pass to the modules?

Please let me know if there is any other info that could help

**Comment by pimvernooij** [comment]
We're looking in to it - is it an option to remove the `count`, `for_each` or `depends_on` argument in the module?

See: https://developer.hashicorp.com/terraform/language/modules/develop/providers

_"A module containing its own provider configurations is not compatible with the for_each, count, and depends_on arguments that were introduced in Terraform v0.13. For more information, see [Legacy Shared Modules with Provider Configurations](https://developer.hashicorp.com/terraform/language/modules/develop/providers#legacy-shared-modules-with-provider-configurations)."_

**Comment by yevhenii0bozhenko** [comment]
Hi @leongraumans , thanks for your input here!
Actually i there is no problem passing variables to components (modules).
The problem is that mach composer is creating additional resource called: `resource "azurerm_app_service_plan" "functionapps"` on which all the azure terraform modules then depend on.
This blocks us declaring providers within modules code.

I've tried using azure-minimal plugin but it still generates the same azurerm_app_service_plan resource.

**Comment by yevhenii0bozhenko** [comment]
This is my main.yaml file for reference:
```
---
mach_composer:
  version: 1
  plugins:
    azure:
      source: mach-composer/azure-minimal
      version: 0.1.0

global:
  environment: dev
  terraform_config:
    azure_remote_state:
      resource_group: test
      storage_account: test
      container_name: test
      state_folder: test
    providers:
      azure: 3.47.0
  cloud: azure
  azure:
    tenant_id: test
    subscription_id: test
    region: westeurope
    resources_prefix: ''

sites:
- identifier: dev
  azure:
    resource_group: test
  components:
  - name: bff
    variables:
      kuber_cluster_name: test
      resource_group_name: test
      container_registry_name: test
      namespace: test
      k8s:
        replicas: 1
        PORT: '3000'
        ALGOLIA_APPLICATION_ID: test
        ALGOLIA_API_KEY: test
        CT_AUTH_URL: test
        CT_PROJECT_KEY: test
        CT_API_URL: test
    secrets:
      CT_CLIENT_ID: test
      CT_SECRET: test
  - name: ui
    variables:
      kuber_cluster_name: test
      resource_group_name: test
      container_registry_name: test
      namespace: test
      k8s:
        replicas: 1
        PORT: '3000'
        NEXT_PUBLIC_GQL_API_URL: test
        NEXT_PUBLIC_ALGOLIA_APPLICATION_ID: test
        NEXT_PUBLIC_ALGOLIA_API_KEY: test

components:
  $ref: "components.yml"

```
And the error on mach-composer init is:
```
Module is incompatible with count, for_each, and depends_on
│
│   on site.tf line 164, in module "ui":
│  164:     azurerm_app_service_plan.functionapps,
│
│ The module at module.ui is a legacy module which contains its own local provider configurations, and so calls to it may not use the count, for_each, or depends_on arguments.
```

**Comment by yevhenii0bozhenko** [comment]
resolved in v2.9.3 with using azure-minimal plugin 0.1.0
